classes/Twig/Block/UpgradeButtonBlock.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Twig\Block;
+
+use Configuration;
+use PrestaShop\Module\AutoUpgrade\ChannelInfo;
+use Twig_Environment;
+use PrestaShop\Module\AutoUpgrade\Upgrader;
+use PrestaShop\Module\AutoUpgrade\Parameters\UpgradeConfiguration;
+use PrestaShop\Module\AutoUpgrade\TaskRunner\AbstractTask;
+use PrestaShop\Module\AutoUpgrade\UpgradeSelfCheck;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator;
+
+class UpgradeButtonBlock
+{
+
+    /**
+     * @var \Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * @var TranslatorComponent
+     */
+    private $translator;
+
+    /**
+     * @var \Upgrader
```

Comment:
\Upgrader or PrestaShop\Module\AutoUpgrade\Upgrader?

Reply:
PrestaShop\Module\AutoUpgrade\Upgrader. The namespace has been added after @eternoendless created these classes.